src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -103,6 +103,17 @@ public static SegmentInfos readSegmentInfos(Directory directory) throws IOExcept
     }
 
     /**
+     * Returns the number of documents in the index referenced by this {@link SegmentInfos}
+     */
+    public static int getNumDocs(SegmentInfos info) {
+        int numDocs = 0;
+        for (SegmentCommitInfo si : info) {
+            numDocs += si.info.getDocCount() - si.getDelCount();
```

Comment:
IndexReader has 2 getters `numDocs` and `maxDocs` the latter excludes deleted docs. If you have two replicas of the same shard you  can only rely on `numDcos` being the same if the seal ID is the same otherwise you relying on merges being deterministic which they are not.